### PR TITLE
Ports: Bump vim port to version 8.2.3800 and fix build on macOS host

### PR DESCRIPTION
**Pull request by unixinspace**
This PR updates vim to 8.2.3800 and adds some flags to package.sh which override the default vim configure (which picks up the host uname values) and provides our own for SerenityOS. This was reported by @BGR360 on the [vim repo](https://github.com/vim/vim/issues/9338).

### CODE REVIEW

File: Ports/vim/package.sh [hunk 1]
```diff
@@ -15,7 +15,9 @@ export vim_cv_tgetent=zero
 export vim_cv_terminfo=yes
 export vim_cv_toupper_broken=no
 export vim_cv_tty_group=world
-
+export vim_cv_uname_output=SerenityOS
+export vim_cv_uname_r_output=1.0-dev
+export vim_cv_uname_m_output=i686
```
Comment (BertalanD) [inline Ports/vim/package.sh:20]: We can query the ports system about the target architecture. Let's use that instead.

```suggestion
export vim_cv_uname_m_output=$SERENITY_ARCH
```

Comment (timschumi) [inline Ports/vim/package.sh:20]: We should use `SERENITY_ARCH` here instead of hardcoding `i686`.

**Review by timschumi — CHANGES_REQUESTED** [review]
Please squash the AvailablePorts.md update commit into the commit that actually changed the package version.

File: Ports/AvailablePorts.md [hunk 2]
```diff
@@ -174,7 +174,7 @@ Please make sure to keep this list up to date when adding and updating ports. :^
 | [`tinycc`](tinycc/)                    | Tiny C Compiler (TinyCC)                                        | dev                      | https://github.com/TinyCC/tinycc                                               |
 | [`tinyscheme`](tinyscheme/)            | TinyScheme Interpreter                                          | 1.42                     | https://sourceforge.net/projects/tinyscheme/                                   |
 | [`tr`](tr/)                            | tr (OpenBSD)                                                    | 6.7                      | https://github.com/ibara/libpuffy                                              |
-| [`vim`](vim/)                          | Vim                                                             | 8.2.2772                 | https://www.vim.org/                                                           |
+| [`vim`](vim/)                          | Vim                                                             | 8.2.3800                 | https://www.vim.org/                                                           |
```
Comment (BertalanD) [inline Ports/AvailablePorts.md:177]: Please squash this commit into the previous one. We generally want every single commit in a PR to pass the CI checks, not just the final one.

To squash these commits:
1. First, remove the commits while keeping their contents intact:
```shell
git reset --soft HEAD~2
````
2. Then, create a new commit from them:
```shell
git commit
````

**Review by BertalanD — CHANGES_REQUESTED** [review]

File: Ports/vim/package.sh (hunk 1 above)
Comment (unixinspace) [reply Ports/vim/package.sh]: Thank you, done!

Comment (unixinspace) [reply Ports/vim/package.sh]: Thank you, done!

File: Ports/AvailablePorts.md (hunk 2 above)
Comment (unixinspace) [reply Ports/AvailablePorts.md:177]: Thank you, that's extremely useful knowledge!

**Review by linusg — CHANGES_REQUESTED** [review]
Please split this into two commits:

- Package version bump
- macOS host build fix (needs some explanation in the commit message)

(not necessarily in that order)

Having `and` in a commit message is often (not always) a sign that the commit does too much.

**Pull request closed without merging**